lib/zclient.c

```diff
@@ -1430,9 +1432,18 @@ int zapi_nexthop_decode(struct stream *s, struct zapi_nexthop *api_nh,
 			   sizeof(struct seg6local_context));
 	}
 
-	if (CHECK_FLAG(api_nh->flags, ZAPI_NEXTHOP_FLAG_SEG6))
-		STREAM_GET(&api_nh->seg6_segs, s,
-			   sizeof(struct in6_addr));
+	if (CHECK_FLAG(api_nh->flags, ZAPI_NEXTHOP_FLAG_SEG6)) {
+		STREAM_GETC(s, api_nh->seg_num);
```

Comment:
I don't see how this won't crash in the future.  In the encoding function just above, there is a test to see `if api_nh->seq_num > 0`  before a write is done.   Which implies that this can and will happen.  So the encoding function *must* write the number of seq_num that are in there so that the reading function *can* properly decode it.

Reply:
Now it should be okay. in  encoder we set seg_num to zero.

Reply:
@pushpasis There was an issue, that Donald pointed out, in initial push that was fixed by deleting the `if api_nh->seg_num >0` statement in the encoding function, such that encoding function must write the number of seq_num (so we cant encode seg_num==zero also). 

So in the initial broken version we had:
- in encoder:
1. ZAPI_NEXTHOP_FLAG_SEG6 is True (set)
2. `if api_nh->seg_num > 0` , this introduced an issue, because when in encoder part seg_num == 0 we didn't encode anything. 
- Decoder on it's part was broken with:
1. ZAPI_NEXTHOP_FLAG_SEG6 is True (set)
2. further decoder was decoding values (seg_num) that were not encoded because of if-statement.